Add Speedometer 3 support to run-benchmark

rniwa opened the pull request:
#### 3d37cb764cf1bf1426080da9325a0548dd8d820e
<pre>
Add Speedometer 3 support to run-benchmark
<a href="https://bugs.webkit.org/show_bug.cgi?id=255752">https://bugs.webkit.org/show_bug.cgi?id=255752</a>

Reviewed by Justin Michaud and Dewei Zhu.

Add a plan file as well as corresponding patches for Speedometer 3.

* Tools/Scripts/webkitpy/benchmark_runner/data/patches/signposts/Speedometer3.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webserver/Speedometer3.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/speedometer3.plan: Added.

Canonical link: <a href="https://commits.webkit.org/263203@main">https://commits.webkit.org/263203@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf8d52efceff0f5d4628002588355c2ddb3d8d04

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3937 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4030 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4141 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5375 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/4192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3915 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4116 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/4013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/4005 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3985 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/4174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3547 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5233 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/3974 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/1675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3522 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/5566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3497 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/3582 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/4993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3989 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/4013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3508 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3522 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3557 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/447 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3788 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

justinmichaud reviewed (approved):
r=me

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/data/plans/speedometer3.plan:
```diff
@@ -0,0 +1,10 @@
+    "webserver_benchmark_patch": "data/patches/webserver/Speedometer3.patch",
+    "signpost_patch": "data/patches/signposts/Speedometer3.patch",
+    "entry_point": "index.html",
+    "output_file": "speedometer3.result"
```
It would be worth running `python3 Tools/Scripts/webkitpy/benchmark_runner/plan_file_optimizer.py Tools/Scripts/webkitpy/benchmark_runner/data/plans/speedometer3.plan` to get `github_subtree` generated so that we can avoid GitHub API rate limit.

dewei-zhu replied:
Ryosuke pointed out `"remote_archive":"https://github.com/WebKit/Speedometer/archive/refs/heads/main.zip",` is from a separate repo, which means we don't need `github_subtree`.

dewei-zhu reviewed (approved):
r=me

rniwa committed:
Add Speedometer 3 support to run-benchmark
https://bugs.webkit.org/show_bug.cgi?id=255752

Reviewed by Justin Michaud and Dewei Zhu.

Add a plan file as well as corresponding patches for Speedometer 3.

* Tools/Scripts/webkitpy/benchmark_runner/data/patches/signposts/Speedometer3.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webserver/Speedometer3.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/speedometer3.plan: Added.

Canonical link: https://commits.webkit.org/263203@main
